common/smoke-test/Initialize-SmokeTests.ps1

```diff
@@ -191,30 +135,136 @@ foreach ($entry in $manifest) {
       $dependencies[$dep] = $packageSpec.dependencies[$dep]
     }
   }
+}
 
+function Start-NewTestResourcesJob {
+  Param(
+    [Parameter(Mandatory = $true)] [System.Collections.Hashtable]$entry,
+    [Parameter(Mandatory = $true)] [string]$baseName,
+    [Parameter(Mandatory = $true)] [string]$resourceGroupName
+  )
 
+  Start-Job -Name $entry.Name -ScriptBlock {
+    &"$using:repoRoot/eng/common/TestResources/New-TestResources.ps1" `
+       -BaseName  $using:baseName `
+       -ResourceGroupName $using:resourceGroupName `
+       -ServiceDirectory $using:entry.ResourcesDirectory `
+       -TestApplicationId $using:TestApplicationId `
+       -TestApplicationSecret $using:TestApplicationSecret `
+       -ProvisionerApplicationId $using:TestApplicationId `
+       -ProvisionerApplicationSecret $using:TestApplicationSecret `
+       -TestApplicationOid $using:TestApplicationOid `
+       -TenantId $using:TenantId `
+       -SubscriptionId $using:SubscriptionId `
+       -Location $using:Location `
+       -Environment $using:Environment `
+       -AdditionalParameters $using:AdditionalParameters `
+       -DeleteAfterHours 24 `
+       -Force `
+       -Verbose `
+       -CI:$using:CI
+  }
 }
 
-Write-Verbose "Writing run-manifest.json"
-($runManifest | ConvertTo-Json -AsArray | Set-Content -Path "$repoRoot/common/smoke-test/run-manifest.json" -Force)
+function Deploy-TestResources {
+  Param([Parameter(Mandatory = $true)] $deployManifest)
 
-Write-Verbose "Writing dependencies into Smoke Test package.json"
-$runnerPackageSpec = Get-Content "$repoRoot/common/smoke-test/package.json" | ConvertFrom-Json -AsHashtable
-$runnerPackageSpec.dependencies = $dependencies
-($runnerPackageSpec | ConvertTo-Json | Set-Content "$repoRoot/common/smoke-test/package.json")
+  $deployedServiceDirectories = @{ }
+  $baseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+  $resourceGroupName = "rg-smoke-$baseName"
+  $dependencies = New-Object 'System.Collections.Generic.Dictionary[string,string]'
+  $runManifest = @()
 
-SetEnvironmentVariable -Name "NODE_PATH" -Value "$PSScriptRoot/node_modules"
+  # Use the same resource group name that New-TestResources.ps1 generates
+  Set-EnvironmentVariable -Name 'AZURE_RESOURCEGROUP_NAME' -Value $resourceGroupName
 
-if ($CI) {
-  # If in CI mark the task as successful even if there are warnings so the
-  # pipeline execution status shows up as red or green
-  Write-Host "##vso[task.complete result=Succeeded; ]DONE"
+  $entryDeployJobs = @()
+
+  # Remove any lingering jobs from this session so output is not polluted
+  Get-Job | Stop-Job
```

Comment:
Will this potentially impact other things outside the context of this script?

Reply:
Based on other feedback below of just using the job list you create I would remove these 2 lines from the script.